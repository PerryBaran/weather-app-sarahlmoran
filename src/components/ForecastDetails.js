import React from "react";

function ForecastDetails({ forecast }) {
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        Date: {new Date(forecast.date).toDateString()}
      </div>
      <div className="forecast-details__humidity">
        Humidity: {forecast.humidity}
      </div>
      <div className="forecast-details__mintemperature">
        Min: {forecast.temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__maxtemperature">
        Max: {forecast.temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__windspeed">
        Wind Speed: {forecast.wind.speed}
      </div>
      <div className="forecast-details__winddirection">
        Wind Direction: {forecast.wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
