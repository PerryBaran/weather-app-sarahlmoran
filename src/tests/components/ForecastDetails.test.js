import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 11111111,
    humidity: 30,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Date: Thu Jan 01 1970")).toHaveClass(
      "forecast-details__date"
    );
    expect(getByText("Humidity: 30")).toHaveClass("forecast-details__humidity");
    expect(getByText("Min: 12°C")).toHaveClass(
      "forecast-details__mintemperature"
    );
    expect(getByText("Max: 22°C")).toHaveClass(
      "forecast-details__maxtemperature"
    );
    expect(getByText("Wind Speed: 13")).toHaveClass(
      "forecast-details__windspeed"
    );
    expect(getByText("Wind Direction: s")).toHaveClass(
      "forecast-details__winddirection"
    );
  });
});
