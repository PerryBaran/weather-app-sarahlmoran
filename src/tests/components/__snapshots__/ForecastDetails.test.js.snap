// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Date: Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30
    </div>
    <div
      class="forecast-details__mintemperature"
    >
      Min: 12°C
    </div>
    <div
      class="forecast-details__maxtemperature"
    >
      Max: 22°C
    </div>
    <div
      class="forecast-details__windspeed"
    >
      Wind Speed: 13
    </div>
    <div
      class="forecast-details__winddirection"
    >
      Wind Direction: s
    </div>
  </div>
</DocumentFragment>
`;
